src/modules/auth/credentials/BulkAssignmentContainer.js

```diff
@@ -0,0 +1,453 @@
+	const [tenantsFilter, setTenantsFilter] = useState("");
+	const [loadingTenants, setLoadingTenants] = useState(true);
+	const [selectedTenants, setSelectedTenants] = useState([{_id: 'Global roles', selectedRole: []}]);
+
+	const [show, setShow] = useState(false);
+	const [showTenantsContentLoader, setShowTenantsContentLoader] = useState(false);
+
+	const resource = "authz:superuser";
+	const resources = useSelector(state => state.auth?.resources);
+
+	// headers for Credentails List
+	const headers = [
+		{
+			name: '',
+			customComponent: {
+				generate: (credential) => (
+					<div>
+						<ButtonWithAuthz
+							title={t(`BulkAssignmentContainer|${credential.assigned ? "Credential already assigned" : "Add credential"}`)}
+							id={credential._id}
+							size="sm"
+							color="primary"
+							outline
+							onClick={() => saveToSelectedCredentials(credential)}
+							resource={resource}
+							resources={resources}
+							disabled={credential.assigned}
+						>
+							<i className="cil-plus"></i>
+						</ButtonWithAuthz>
+					</div>
+				)
+			}
+		},
+		{
+			name: '',
+			customComponent: {
+				generate: (obj) => (
+					<div className="no-wrap-40ch">
+						{obj.suspended === true ?
+							<span className="cil-user-unfollow text-muted mr-1" title={(obj.registered === false) ? t("BulkAssignmentContainer|Credentials invited") : t("BulkAssignmentContainer|Credentials suspended")}/>
+							: <span className="cil-user mr-1" />}
+						<Link
+							style={{color: obj.suspended === true && '#73818f'}}
+							onClick={() => confirmForm(`/auth/credentials/${obj._id}`)}
+							>
+							{/* TODO: substitute with Credentials component, when it's ready */}
+							{obj.username ?? obj._id}
+						</Link>
+					</div>
+				)
+			},
+		}
+	];
+
+	// headers for Tenants List
+	const tenantHeaders = [
+		{
+			name: '',
+			customComponent: {
+				generate: (tenant) => (
+					<div>
+						<ButtonWithAuthz
+							title={t(`BulkAssignmentContainer|${tenant.assigned ? "Tenant already assigned" : "Add tenant"}`)}
+							id={tenant._id}
+							size="sm"
+							color="primary"
+							outline
+							onClick={() => saveToSelectedTenants(tenant)}
+							resource={resource}
+							resources={resources}
+							disabled={tenant.assigned}
+						>
+							<i className="cil-plus"></i>
+						</ButtonWithAuthz>
+					</div>
+				)
+			}
+		},
+		{
+			name: '',
+			customComponent: {
+				generate: (tenant) => (
+					<Link onClick={() => confirmForm(`/auth/tenant/${tenant._id}`)}>
+						{tenant._id}
+					</Link>
+				)
+			}
+		}
+	];
+
+	// UseEffect to fetch data for Credentials List based on changes in page/credentialsFilter/limit
+	useEffect(() => {
+		setShow(false);
+		if (data?.length === 0) {
+			// Timeout delays appearance of content loader in DataTable. This measure prevents 'flickering effect' during fast fetch of data, where content loader appears just for a split second.
```

Comment:
@petrKavulok As I wrote somewhere else before, I think its ok, when content loader is displayed even for a milisecond. You are playing too much with the performance here.